src/Blueprint/Blueprint/Cmdlets/BlueprintCmdletBase.cs

```diff
@@ -0,0 +1,382 @@
+
+            return servicePrincipal;
+        }
+
+        /// <summary>
+        /// Assign owner role to Blueprint RP (so that we can do deployments)
+        /// </summary>
+        /// <param name="subscriptionId"></param>
+        /// <param name="servicePrincipal"></param>
+        protected void AssignOwnerPermission(string subscriptionId, ServicePrincipal servicePrincipal)
+        {
+            string scope = string.Format(BlueprintConstants.SubscriptionScope, subscriptionId);
+
+            var filter = new Rest.Azure.OData.ODataQuery<RoleAssignmentFilter>();
+            filter.SetFilter(a => a.AssignedTo(servicePrincipal.ObjectId));
+
+            var roleAssignmentList = AuthorizationManagementClient.RoleAssignments.ListForScopeAsync(scope, filter).GetAwaiter().GetResult();
+
+            var roleAssignment = roleAssignmentList?
+                .Where(ra => ra.Id.EndsWith(BlueprintConstants.OwnerRoleDefinitionId))
+                .FirstOrDefault();
+
+            if (roleAssignment != null) return;
+
+            var roleAssignmentParams = new RoleAssignmentProperties(
+                roleDefinitionId: BlueprintConstants.OwnerRoleDefinitionId, principalId: servicePrincipal.ObjectId);
+
+            try
+            {
+                AuthorizationManagementClient.RoleAssignments.CreateAsync(scope: scope, 
+                    roleAssignmentName: Guid.NewGuid().ToString(), 
+                    parameters: new RoleAssignmentCreateParameters(roleAssignmentParams))
+                    .GetAwaiter().GetResult();
+            }
+            catch (Exception ex)
+            {
+                // ignore if it already exists
+                if (ex is CloudException cex && cex.Response.StatusCode != System.Net.HttpStatusCode.Conflict)
+                {
+                    throw new CloudException(cex.Message);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Get management group ancestors for a given subscription
+        /// </summary>
+        /// <param name="subscriptionId"></param>
+        /// <returns></returns>
+        protected List<string> GetManagementGroupAncestorsForSubscription(string subscriptionId)
+        {
+            List<string> managementGroupAncestors = new List<string>();
+
+            if (subscriptionId != null)
+            {
+                string result = GetManagementGroupAncestorsAsync(subscriptionId).GetAwaiter().GetResult();
+                var resultJObjects = JObject.Parse(result);
+                var managementGroupAncestorsObjects = resultJObjects["managementGroupAncestors"].Children().ToList();
+
+                foreach (var mgObject in managementGroupAncestorsObjects)
+                {
+                    managementGroupAncestors.Add(mgObject.ToString());
+                }
+
+                if (managementGroupAncestors.Count > 0)
+                {
+                    // Remove the last item which is the tenantId.
```

Comment:
>tenantId [](start = 57, length = 8)

It's not. The last item is the root MG whose ID is the same guid as the tenant ID.

Reply:
Thanks for catching that. I'll update the code.

---
In reply to: [257093763](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257093763) [](ancestors = 257093763)